core/chains/evm/client/simulated_backend_client.go

```diff
@@ -64,37 +73,82 @@ func (c *SimulatedBackendClient) checkEthCallArgs(
 			"must be an eth.CallArgs, got %+#v", args[0])
 	}
 	blockNumber, err := c.blockNumber(args[1])
-	if err != nil || blockNumber.Cmp(c.currentBlockNumber()) != 0 {
+	if err != nil {
 		return nil, nil, fmt.Errorf("fourth arg to SimulatedBackendClient.Call "+
-			"must be the string \"latest\", or a *big.Int equal to current "+
-			"blocknumber, got %#+v", args[1])
+			"must be the string \"latest\", or a *big.Int, got %#+v", args[1])
+	}
+
+	toAddr, err := interfaceToAddress(callArgs["to"])
+	if err != nil {
+		return nil, nil, err
 	}
+
+	// to and from need to map to a common.Address but could come in as a string
```

Comment:
nit: comment unaligned.
Maybe move the toAddr and fromAddr parsing before setting CallArgs, and put this comment there.